### ISSUE TITLE
Make version of R packages explicit in R_dependencies.R

### ISSUE THREAD

**Issue by kurtwheeler** [AlexsLemonade/refinebio#71]
I meant to do this when setting up the workers' Docker image, but I guess I got so happy everything actually worked that I forgot to do so. It doesn't necessarily seem easy, but @jaclyn-taroni has helpfully provided the following R code which allows brainarray packages' versions to be specified:

```R
InstallBrainarray <- function(platform, org.code, ba.version) {
    # This function makes use of devtools::install_url to install Brainarray
    # packages for the annotation of Affymetrix data. Specifically, the packages
    # required for use with SCAN.UPC and affy (RMA) are installed.
    #
    # Args:
    #   platform: The Affymetrix platform for which brainarray
    #             packages are to be installed (e.g., "hgu133plus2")
    #   org.code: Two letter organism code -- human would be "hs"
    #        ba.version: What version of brainarray should be used? (e.g., "21.0.0")
    #
    #    Returns:
    #      NULL - this package completes installation of these packages and does not
    #                     return any values 

    # make sure platform and org.code are all lowercase and lack punctuation
    platform <- tolower(gsub("[[:punct:]]", "", platform))
    org.code <- tolower(gsub("[[:punct:]]", "", org.code))

    # probe version for use with SCAN.UPC
    probe.pkg.name <- paste0(platform, org.code, "entrezgprobe_",
                                         ba.version, ".tar.gz")
    probe.url <- paste0("http://mbni.org/customcdf/21.0.0/entrezg.download/",
                                            probe.pkg.name)
    devtools::install_url(probe.url)

    # cdf version for use with affy::RMA
    cdf.pkg.name <- paste0(platform, org.code, "entrezgcdf_",
                                         ba.version, ".tar.gz")
    cdf.url <- paste0("http://mbni.org/customcdf/21.0.0/entrezg.download/",
                                        cdf.pkg.name)
    devtools::install_url(cdf.url)


}

#### install brainarray main ---------------------------------------------------

# HGU133Plus2
InstallBrainarray(platform = "hgu133plus2",
                                    org.code = "hs",
                                    ba.version = "21.0.0")
```

**Comment by jaclyn-taroni** [comment]
Worth noting that I made a error -- the version is also specified in the URL. Also, note that we will not need the `entrezgcdf` packages, as we do not plan to use RMA normalization at the moment.

**Comment by jaclyn-taroni** [comment]
We might want to use devtools for this: https://cran.r-project.org/web/packages/devtools/README.html

**Comment by kurtwheeler** [comment]
Took a look through devtools a bit:

> install_version() installs a specified version from cran.
That means we can lock down the version number easily for cran packages.

> install_github() from github
Has a [ref parameter](https://github.com/r-lib/devtools/blob/master/R/install-github.r#L54) which will allow us to point to a specific commit hash, which we can then document which version that corresponds to.

> install_url() from an arbitrary url
Should be sufficient for us to install a brainarray package of a specific version because as noted above the version can be specified for those URLs.

Bioconductor is a bit weirder because it seems to be intended to be used an ecosystem and doesn't really seem to attempt to support usage of its packages outside of a fairly specific workflow. However, according to https://support.bioconductor.org/p/28238/:
> each version of bioconductor is meant to be used with a specific version of R
and apparently will install packages which are compatible with the version of bioconductor which is installed, which apparently will be automagically figured out by the version of R which is installed. So I think that within `r_dependencies.R` we can specify the version of R to install, figure out what version of Bioconductor that corresponds to, and then figure out what versions of the packages we want will be installed. Perhaps with further investigation a more programmatic method can be discovered, but that does seem a bit manually intensive.

I have found https://bioconductor.org/checkResults/ which has links for all bioconductor releases and each release has the version of each package associated with that release.

**Comment by dongbohu** [comment]
Fixed in PR #176
